sympy/stats/tests/test_stochastic_process.py

```diff
@@ -194,3 +198,155 @@ def test_BernoulliProcess():
+
+    assert P((X(t)< 2) & (X(d)> 3), Contains(X(t), Interval.Lopen(0, 2))
+        & Contains(X(d), Interval.Ropen(2, 4))) == Probability((X(d) > 3) & (X(t) < 2),
+        Contains(X(d), Interval.Ropen(2, 4)) & Contains(X(t), Interval.Lopen(0, 2)))
+
+    assert str(P(Not((X(t) < 5) & (X(d) > 3)), Contains(X(t), Interval.Ropen(2, 4)) &
+        Contains(X(d), Interval.Lopen(7, 8))).simplify()) == \
+                '-(1 - erf(3*sqrt(2)/2))*(2 - erfc(5/2))/4 + 1'
+    assert E(X(t)) == 0
+    assert E(x*(X(t) + X(d))*(X(t)**2+X(d)**2), Contains(X(t), Interval.Lopen(0, 1))
+    & Contains(X(d), Interval.Ropen(1, 2))) == Expectation(x*(X(d) + X(t))*(X(d)**2 + X(t)**2),
+    Contains(X(d), Interval.Ropen(1, 2)) & Contains(X(t), Interval.Lopen(0, 1)))
+    assert E(X(t) + x*E(X(3))) == 0
+
+
+def test_GammaProcess_symbolic():
+    t, d, x, y, g, l = symbols('t d x y g l', positive=True)
+    X = GammaProcess("X", l, g)
+
+    raises(NotImplementedError, lambda: X[t])
+    raises(IndexError, lambda: X(-1))
+    assert isinstance(X(t), RandomIndexedSymbol)
+
+    assert X.distribution(X(t)) == GammaDistribution(g*t, 1/l)
+    assert X.joint_distribution(5, X(3)) == JointDistributionHandmade(Lambda(
+        (X(5), X(3)), l**(8*g)*exp(-l*X(3))*exp(-l*X(5))*X(3)**(3*g - 1)*X(5)**(5*g
+        - 1)/(gamma(3*g)*gamma(5*g))))
+    assert E(X(t)) == g*t/l
+    assert variance(X(t)).simplify() == g*t/l**2
+    assert E(X(t)**2 + X(d)*2 + X(y)**3, Contains(X(t), Interval.Lopen(0, 1))
+        & Contains(X(d), Interval.Lopen(1, 2)) & Contains(X(y), Interval.Ropen(3, 4))) == \
+            2*g/l + (g**2 + g)/l**2 + (g**3 + 3*g**2 + 2*g)/l**3
+
+    assert P(X(t) > 3, Contains(X(t), Interval.Lopen(3, 4))).simplify() == \
+                                1 - lowergamma(g, 3*l)/gamma(g)
+
+def test_GammaProcess_numeric():
+    t, d, x, y = symbols('t d x y', positive=True)
+    X = GammaProcess("X", 1, 2)
+    assert X.state_space == Interval(0, oo)
+    assert X.index_set == Interval(0, oo)
+    assert X.lamda == 1
+    assert X.gama == 2
+
+    raises(ValueError, lambda: GammaProcess("X", -1, 2))
+    raises(ValueError, lambda: GammaProcess("X", 0, -2))
+    raises(ValueError, lambda: GammaProcess("X", -1, -2))
+
+    assert P((X(t) > 4) & (X(d) > 3) & (X(x) > 2) & (X(y) > 1), Contains(X(t),
+        Interval.Lopen(0, 1)) & Contains(X(d), Interval.Lopen(1, 2)) & Contains(X(x),
+        Interval.Lopen(2, 3)) & Contains(X(y), Interval.Lopen(3, 4))).simplify() == \
+                                                            120*exp(-10)
+    assert P(Not((X(t) < 5) & (X(d) > 3)), Contains(X(t), Interval.Ropen(2, 4)) &
+        Contains(X(d), Interval.Lopen(7, 8))).simplify() == -4*exp(-3) + 472*exp(-8)/3 + 1
+    assert P((X(t) > 2) | (X(t) < 4), Contains(X(t), Interval.Ropen(1, 4))).simplify() == \
+                                            -643*exp(-4)/15 + 109*exp(-2)/15 + 1
+
+    assert E(X(t)) == 2*t
+    assert E(X(2) + x*E(X(5))) == 10*x + 4
```

Comment:
Can you provide some references for the tests? It will be easy to verify them.

Reply:
For symbolic gamma process, I used [https://en.wikipedia.org/wiki/Gamma_process](https://en.wikipedia.org/wiki/Gamma_process) to verify, and as it is a Levy Process, see: [https://en.wikipedia.org/wiki/L%C3%A9vy_process](https://en.wikipedia.org/wiki/L%C3%A9vy_process), I used the tests similar to PoissonProcess

Reply:
Can you provide the references from where you took these tests, like are these tests examples of some lecture notes or book? Or did you think of them on your own?

Reply:
For verifying E, variance symbolically, I used Wikipedia, and the remaining ones, I thought and framed according to Poisson Process, as I didn't find problems on Gamma Process.

Reply:
Can you add some comments directly into the code describing the computations needed to verify the tests?